app/logic/solana_logic.rb

```diff
@@ -651,9 +653,15 @@ def check_epoch
 
   def existing_validators(validator_attrs, network)
     accounts = validator_attrs.map { |validator| validator["identityPubkey"] }
-    db_validator_accounts = Validator.where(network: network, account: accounts).pluck(:account)
-    validator_attrs.select do |validator|
-      db_validator_accounts.include?(validator["identityPubkey"])
-    end
+    db_validator_accounts = Validator.where(network: network, account: accounts)
```

Comment:
I don't think those changes are necessary - there's no reason why we should use ids here

Reply:
We need to extract those ids in order to add validator relation to validator history in `validator_history_update` method.